cdap-gateway/src/main/java/co/cask/cdap/gateway/router/handlers/HttpRequestHandler.java

```diff
@@ -157,6 +158,14 @@ private HttpRequest applyProxyRules(HttpRequest request) {
   public void exceptionCaught(ChannelHandlerContext ctx, ExceptionEvent e)  {
     Throwable cause = e.getCause();
 
+    // avoid handling exception more than once from a handler, to avoid a possible infinite recursion
+    if (exceptionHandled) {
+      LOG.error("Not handling exception due to already having handled an exception in Request Handler {}",
+                ctx.getChannel(), cause);
+      return;
+    }
+    exceptionHandled = true;
```

Comment:
Hmm, if the LOG statement throws an exception, then the `exceptionCaught` method will get called again?